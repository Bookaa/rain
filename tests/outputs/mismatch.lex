keyword 'import'
name 'except'
newline
name 'none'
symbol '='
keyword 'func'
symbol '('
symbol ')'
indent
name 'print'
symbol '('
string 'no args'
symbol ')'
newline
dedent
newline
name 'one'
symbol '='
keyword 'func'
symbol '('
name 'a'
symbol ')'
indent
name 'print'
symbol '('
string 'one arg'
symbol ')'
newline
dedent
newline
name 'two'
symbol '='
keyword 'func'
symbol '('
name 'a'
symbol ','
name 'b'
symbol ')'
indent
name 'print'
symbol '('
string 'two args'
symbol ')'
newline
dedent
newline
name 'main'
symbol '='
keyword 'func'
symbol '('
symbol ')'
indent
keyword 'let'
name 'a'
symbol '='
int 1
newline
keyword 'let'
name 'b'
symbol '='
string 'str'
newline
keyword 'catch'
name 'r'
indent
name 'a'
symbol '('
name 'b'
symbol ')'
newline
dedent
newline
keyword 'if'
operator '!'
name 'except'
symbol '.'
name 'safe'
symbol '('
name 'r'
symbol ')'
indent
name 'print'
symbol '('
name 'r'
symbol '.'
name 'msg'
symbol ')'
newline
dedent
newline
keyword 'catch'
name 'r'
indent
keyword 'for'
name 'x'
keyword 'in'
'null'
indent
name 'print'
symbol '('
name 'x'
symbol ')'
newline
dedent
newline
dedent
newline
keyword 'if'
operator '!'
name 'except'
symbol '.'
name 'safe'
symbol '('
name 'r'
symbol ')'
indent
name 'print'
symbol '('
name 'r'
symbol '.'
name 'msg'
symbol ')'
newline
dedent
newline
keyword 'catch'
name 'r'
indent
name 'none'
symbol '('
symbol ')'
newline
name 'none'
symbol '('
int 1
symbol ')'
newline
dedent
newline
keyword 'if'
operator '!'
name 'except'
symbol '.'
name 'safe'
symbol '('
name 'r'
symbol ')'
indent
name 'print'
symbol '('
name 'r'
symbol '.'
name 'msg'
symbol ')'
newline
dedent
newline
keyword 'catch'
name 'r'
indent
name 'one'
symbol '('
int 1
symbol ')'
newline
name 'one'
symbol '('
symbol ')'
newline
dedent
newline
keyword 'if'
operator '!'
name 'except'
symbol '.'
name 'safe'
symbol '('
name 'r'
symbol ')'
indent
name 'print'
symbol '('
name 'r'
symbol '.'
name 'msg'
symbol ')'
newline
dedent
newline
keyword 'catch'
name 'r'
indent
name 'two'
symbol '('
int 1
symbol ','
int 2
symbol ')'
newline
name 'two'
symbol '('
int 1
symbol ','
int 2
symbol ','
int 3
symbol ')'
newline
dedent
newline
keyword 'if'
operator '!'
name 'except'
symbol '.'
name 'safe'
symbol '('
name 'r'
symbol ')'
indent
name 'print'
symbol '('
name 'r'
symbol '.'
name 'msg'
symbol ')'
newline
dedent
newline
dedent
newline
EOF
