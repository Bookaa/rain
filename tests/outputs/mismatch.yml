!program;1
stmts:
- !import;1
  name: except
  rename: null
- !assn;1
  let: true
  lhs: !name;1
    value: none
  rhs: !func;1
    body: !block;1
      stmts:
      - !call;1
        args:
        - !str;1
          value: no args
        catch: false
        func: !name;1
          value: print
    params: []
- !assn;1
  let: true
  lhs: !name;1
    value: one
  rhs: !func;1
    body: !block;1
      stmts:
      - !call;1
        args:
        - !str;1
          value: one arg
        catch: false
        func: !name;1
          value: print
    params:
    - a
- !assn;1
  let: true
  lhs: !name;1
    value: two
  rhs: !func;1
    body: !block;1
      stmts:
      - !call;1
        args:
        - !str;1
          value: two args
        catch: false
        func: !name;1
          value: print
    params:
    - a
    - b
- !assn;1
  let: true
  lhs: !name;1
    value: main
  rhs: !func;1
    body: !block;1
      stmts:
      - !assn;1
        let: true
        lhs: !name;1
          value: a
        rhs: !int;1
          value: 1
      - !assn;1
        let: true
        lhs: !name;1
          value: b
        rhs: !str;1
          value: str
      - !catch;1
        body: !block;1
          stmts:
          - !call;1
            args:
            - !name;1
              value: b
            catch: false
            func: !name;1
              value: a
        name: r
      - !if;1
        body: !block;1
          stmts:
          - !call;1
            args:
            - !index;1
              lhs: !name;1
                value: r
              rhs: !str;1
                value: msg
            catch: false
            func: !name;1
              value: print
        els: null
        pred: !unary;1
          op: '!'
          val: !call;1
            args:
            - !name;1
              value: r
            catch: false
            func: !index;1
              lhs: !name;1
                value: except
              rhs: !str;1
                value: safe
      - !catch;1
        body: !block;1
          stmts:
          - !for;1 {}
        name: r
      - !if;1
        body: !block;1
          stmts:
          - !call;1
            args:
            - !index;1
              lhs: !name;1
                value: r
              rhs: !str;1
                value: msg
            catch: false
            func: !name;1
              value: print
        els: null
        pred: !unary;1
          op: '!'
          val: !call;1
            args:
            - !name;1
              value: r
            catch: false
            func: !index;1
              lhs: !name;1
                value: except
              rhs: !str;1
                value: safe
      - !catch;1
        body: !block;1
          stmts:
          - !call;1
            args: []
            catch: false
            func: !name;1
              value: none
          - !call;1
            args:
            - !int;1
              value: 1
            catch: false
            func: !name;1
              value: none
        name: r
      - !if;1
        body: !block;1
          stmts:
          - !call;1
            args:
            - !index;1
              lhs: !name;1
                value: r
              rhs: !str;1
                value: msg
            catch: false
            func: !name;1
              value: print
        els: null
        pred: !unary;1
          op: '!'
          val: !call;1
            args:
            - !name;1
              value: r
            catch: false
            func: !index;1
              lhs: !name;1
                value: except
              rhs: !str;1
                value: safe
      - !catch;1
        body: !block;1
          stmts:
          - !call;1
            args:
            - !int;1
              value: 1
            catch: false
            func: !name;1
              value: one
          - !call;1
            args: []
            catch: false
            func: !name;1
              value: one
        name: r
      - !if;1
        body: !block;1
          stmts:
          - !call;1
            args:
            - !index;1
              lhs: !name;1
                value: r
              rhs: !str;1
                value: msg
            catch: false
            func: !name;1
              value: print
        els: null
        pred: !unary;1
          op: '!'
          val: !call;1
            args:
            - !name;1
              value: r
            catch: false
            func: !index;1
              lhs: !name;1
                value: except
              rhs: !str;1
                value: safe
      - !catch;1
        body: !block;1
          stmts:
          - !call;1
            args:
            - !int;1
              value: 1
            - !int;1
              value: 2
            catch: false
            func: !name;1
              value: two
          - !call;1
            args:
            - !int;1
              value: 1
            - !int;1
              value: 2
            - !int;1
              value: 3
            catch: false
            func: !name;1
              value: two
        name: r
      - !if;1
        body: !block;1
          stmts:
          - !call;1
            args:
            - !index;1
              lhs: !name;1
                value: r
              rhs: !str;1
                value: msg
            catch: false
            func: !name;1
              value: print
        els: null
        pred: !unary;1
          op: '!'
          val: !call;1
            args:
            - !name;1
              value: r
            catch: false
            func: !index;1
              lhs: !name;1
                value: except
              rhs: !str;1
                value: safe
    params: []
