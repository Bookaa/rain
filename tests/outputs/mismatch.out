uncallable value
uncallable value
no args
argument mismatch
one arg
argument mismatch
two args
argument mismatch
